modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernelBase.C

```diff
@@ -0,0 +1,108 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZMInterfaceKernelBase.h"
+
+InputParameters
+CZMInterfaceKernelBase::validParams()
+{
+  InputParameters params = InterfaceKernel::validParams();
+  params.addRequiredParam<unsigned int>("component",
+                                        "the component of the "
+                                        "displacement vector this kernel is working on:"
+                                        " component == 0, ==> X"
+                                        " component == 1, ==> Y"
+                                        " component == 2, ==> Z");
+  params.set<bool>("_use_undisplaced_reference_points") = true;
```

Comment:
what is this `_use_undisplaced_reference_points`?

Reply:
this is a parameter allowing to use fake neighbors when a kernel is used with 'displaced_mesh=True'. At the moment all CZMinterface kernel use 'displaced_mesh=false', but it should stay there until fake neighbors are not part of libmesh